Sources/Maintenance/Maintenance.swift

```diff
@@ -0,0 +1,124 @@
+    /// minimum fetch interval expires. Note that you can override this default on a per-fetch request
+    /// basis using `RemoteConfig.fetch(withExpirationDuration:)`. For example, setting
+    /// the expiration duration to 0 in the fetch request will override the `minimumFetchInterval` and
+    /// allow the request to proceed. For more information, see [Firebase Doccumentation]( https://firebase.google.com/docs/remote-config/get-started?platform=ios#throttling)
+    ///   - defaultValues: Sets config defaults for parameter keys and values in the default namespace config by using A dictionary mapping a String * key to a Any * value.
+    /// - Returns: ``Maintenance/StatusType`` and ``Maintenance/Detail``
+    /// - Throws: ``Helpers/CustomError``
+    public static func fetchFromFirebaseRemoteConfige(minimumFetchInterval: Double? = nil,
+                                                      bannerKey: String,
+                                                      fullscreenKey: String,
+                                                      bannerDetailsKey: String,
+                                                      fullscreenDetailsKey: String) async throws -> (StatusType, Detail)? {
+
+        let remoteConfig = RemoteConfig.remoteConfig()
+
+        RemoteConfigHelper.customizeSettingsIfNeeded(
+            remoteConfig: remoteConfig,
+            minimumFetchInterval: minimumFetchInterval)
+        
+        let fetchedRemoteConfig = await RemoteConfigHelper.fetchRemoteConfig(remoteConfig: remoteConfig)
+        
+        let activatedRemoteConfig = try await RemoteConfigHelper.activeRemoteConfig(remoteConfig: fetchedRemoteConfig.get())
+        
+        let banner = try RemoteConfigHelper.getBoolValueForKey(
+            remoteConfig: activatedRemoteConfig.get(),
+            key: bannerKey)
+        
+        let fullscreen = try RemoteConfigHelper.getBoolValueForKey(
+            remoteConfig: activatedRemoteConfig.get(),
+            key: fullscreenKey)
+        
+        let bannerDetail = try RemoteConfigHelper.getModeledValueForKey(
+            remoteConfig: activatedRemoteConfig.get(),
+            key: bannerDetailsKey,
+            expectationModel: Detail.self)
+        
+        let fullscreenDetail = try RemoteConfigHelper.getModeledValueForKey(
+            remoteConfig: activatedRemoteConfig.get(),
+            key: fullscreenDetailsKey,
+            expectationModel: Detail.self)
+        
+        return isThereAnyMaintenance(
+            fullscreen: fullscreen,
+            banner: banner,
+            fullscreenDetail: fullscreenDetail,
+            bannerDetail: bannerDetail)
+        
+        
+    }
+    
+    static func isThereAnyMaintenance(fullscreen: Bool,
+                                      banner: Bool,
+                                      fullscreenDetail: Detail,
+                                      bannerDetail: Detail) -> (StatusType, Detail)? {
+        
```

Comment:
For me its very hard to read the return type of this function, I think this more readable so we have the -> on its own line so you dont miss the return type. I was thinking this was a void since i didnt see the ->.

I think our linting tool also prefers this way right?

    static func isThereAnyMaintenance(
        fullscreen: Bool,
        banner: Bool,
        fullscreenDetail: Detail,
        bannerDetail: Detail
    ) -> (StatusType, Detail)? {
        
        if fullscreen {
            return (.fullscreen, fullscreenDetail)
        }
        
        if banner {
            return (.banner, bannerDetail)
        }
        
        return nil
    }